app/src/common-logging/common-logging-http.js

```diff
@@ -0,0 +1,75 @@
+const axios = require('axios');
+const oauth = require('axios-oauth-client');
+const tokenProvider = require('axios-token-interceptor');
+
+class AuthorizedConnection {
+  constructor(options) {
+    if (!options || !options.tokenUrl || !options.clientId || !options.clientSecret) {
+      throw new Error('AuthorizedConnection is not configured.  Check configuration.');
+    }
+
+    this.tokenUrl = options.tokenUrl;
+
+    this.axios = axios.create();
+    this.axios.interceptors.request.use(
+      // Wraps axios-token-interceptor with oauth-specific configuration,
+      // fetches the token using the desired claim method, and caches
+      // until the token expires
+      oauth.interceptor(tokenProvider, oauth.client(axios.create(), {
+        url: this.tokenUrl,
+        grant_type: 'client_credentials',
+        client_id: options.clientId,
+        client_secret: options.clientSecret,
+        scope: ''
+      }))
+    );
+  }
+}
+
+class CommonLoggingHttp {
+  constructor(options) {
+    if (!options || !options.tokenUrl || !options.clientId || !options.clientSecret || !options.apiUrl) {
+      throw new Error('CommonLoggingHttp is not configured.  Check configuration.');
+    }
+    this.connection = new AuthorizedConnection(options);
+    this.axios = this.connection.axios;
+    this.apiUrl = options.apiUrl;
+  }
+
+  async xfer(messages) {
+    if (messages) {
+      const batch = Array.isArray(messages) ? messages : [messages];
+
+      if (!batch.length) {
+        return -2;
```

Comment:
It might not be a bad idea to have these negative number values be specified in a constants file - otherwise they're just potentially magic numbers without too much context as to what they represent.

Reply:
They are really only there for testing purposes.
We can deal when/if we move to a npm package.